src/utils/parse.rs

```diff
@@ -0,0 +1,184 @@
+use crate::{model::prelude::*, prelude::*};
+
+/// Parse a value from a string in context of a received message.
+///
+/// This trait is a superset of [`std::str::FromStr`] (the trait used in [`str::parse`]). The
```

Comment:
Users can refer to `FromStr`'s documentation to discover where it is used, no need to explain it here.
```suggestion
/// This trait is a superset of [`std::str::FromStr`]. The
```